lib/signaling/v2/peerconnection.js

```diff
@@ -450,9 +449,15 @@ PeerConnectionV2.prototype._setLocalDescription = function _setLocalDescription(
 
   return Promise.resolve().then(function() {
     if (description.sdp) {
-      description.sdp = isChrome && vp8SimulcastRequested
-        ? self._setSimulcast(description.sdp, self._trackIdsToAttributes)
-        : description.sdp;
+      // NOTE(mmalavalli): We do not directly modify "description.sdp" here as
+      // "description" might be an RTCSessionDescription, in which case its
+      // properties are immutable.
+      description = {
+        type: description.type,
+        sdp: isChrome && vp8SimulcastRequested
+          ? self._setSimulcast(description.sdp, self._trackIdsToAttributes)
+          : description.sdp
+      };
```

Comment:
LGTM—should we do the same in `_setRemoteDescription`? I see it modifies `description.sdp`

Reply:
@markandrus In `_setRemoteDescription` we only get the dict because usually we are passing the contents of the RSP message. I don't think there is any case where we would be passing an RTCSessionDescription to it.